lib/manipulation.ex

```diff
@@ -638,98 +642,50 @@ defmodule LicensePlatePT.Manipulation do
     end
   end
 
-  @type1_invalid_letters [
-    "AM",
-    "AP",
-    "CC",
-    "EM",
-    "MB",
-    "MC",
-    "ME",
-    "MF",
-    "MG",
-    "MH",
-    "MI",
-    "MJ",
-    "ML",
-    "MU",
-    "MV",
-    "MX",
-    "MZ",
-    "NU",
-    "NV",
-    "NZ",
-    "OU",
-    "OV",
-    "OZ",
-    "PR",
-    "PU",
-    "PV",
-    "PZ",
-    "QU",
-    "QV",
-    "QZ",
-    "RU",
-    "RV",
-    "RZ",
-    "SU",
-    "SV",
-    "SZ",
-    "TB",
-    "TC",
-    "TD",
-    "TE",
-    "TF",
-    "TG",
-    "TH",
-    "TJ",
-    "TL",
-    "TQ",
-    "TZ",
-    "UM",
-    "UN",
-    "UO",
-    "UP",
-    "UQ",
-    "UR",
-    "US",
-    "UT",
-    "UV",
-    "VM",
-    "VN",
-    "VO",
-    "VP",
-    "VQ",
-    "VR",
-    "VS",
-    "VT",
-    "ZA",
-    "ZC",
-    "ZD",
-    "ZG",
-    "ZH",
-    "ZI",
-    "ZJ",
-    "ZL",
-    "ZM",
-    "ZM",
-    "ZQ",
-    "ZT",
-    "ZU",
-    "ZV"
-  ]
+  defguardp same_letters(l1, l2, l3, l4) when l1 == l2 and l2 == l3 and l3 == l4
+
+  @vowels ["A", "E", "I", "O", "U"]
+  defguardp contains_vowels_edge_cases(l1, l2, l3, l4)
```

Comment:
might be better to resign from the idea of using guards for the sake of readability - this is very hard to read.